ReactiveFeedbackTests/SystemTests.swift

```diff
@@ -181,4 +181,39 @@ class SystemTests: XCTestCase {
         expect(value) == "initial_a"
         expect(startCount) == 2
     }
+
+    func test_should_not_miss_delivery_to_reducer_when_started_asynchronously() {
+        let creationScheduler = QueueScheduler()
+        let systemScheduler = QueueScheduler()
+
+        let observedState: Atomic<[String]> = Atomic([])
+
+        let semaphore = DispatchSemaphore(value: 0)
```

Comment:
Pardon my ignorance, but why do we need the aid of semaphore for this test?

Reply:
Reproducing scenarios that require specific timing order often require manual synchronisation and sometimes repeated runs, because the OS scheduler need not do you a favour to launch every work item at the same time you expected.